packages/types/src/lib/models.ts

```diff
@@ -0,0 +1,170 @@
+import { SiweMessage } from 'siwe';
+import { AuthSig } from './interfaces';
+
+export type PlainJSON =
+  | boolean
+  | number
+  | string
+  | { [key: string]: PlainJSON }
+  | Array<PlainJSON>;
+export type AttenuationsObject = {
+  [key: string]: { [key: string]: Array<PlainJSON> };
+};
+export type CID = string;
+
+/**
+ * These are the user-facing abilities that can be granted to a session.
+ */
+export enum LitAbility {
+  /**
+   * This is the ability to process an encryption access control condition.
+   * The resource will specify the corresponding hashed key value of the
+   * access control condition.
+   */
+  AccessControlConditionDecryption = 'access-control-condition-decryption',
+
+  /**
+   * This is the ability to process a signing access control condition.
+   * The resource will specify the corresponding hashed key value of the
+   * access control condition.
+   */
+  AccessControlConditionSigning = 'access-control-condition-signing',
+
+  /**
+   * This is the ability to use a PKP for signing purposes. The resource will specify
+   * the corresponding PKP token ID.
+   */
+  PKPSigning = 'pkp-signing',
+
+  /**
+   * This is the ability to use a Rate Limit Increase (Capacity Credits NFT) token during
+   * authentication with the nodes. The resource will specify the corresponding
+   * Capacity Credits NFT token ID.
+   */
+  RateLimitIncreaseAuth = 'rate-limit-increase-auth',
+
+  /**
+   * This is the ability to execute a Lit Action. The resource will specify the
+   * corresponding Lit Action IPFS CID.
+   */
+  LitActionExecution = 'lit-action-execution',
+}
+/**
+ * Prefixes used for identifying various LIT resources.
+ *
+ * @description These resource prefixes are also used as valid IRI schemes.
+ */
+export enum LitResourcePrefix {
+  AccessControlCondition = 'lit-accesscontrolcondition',
+  PKP = 'lit-pkp',
+  RLI = 'lit-ratelimitincrease',
+  LitAction = 'lit-litaction',
+}
+
+export interface ISessionCapabilityObject {
```

Comment:
This should be available in the ReCap package?

Reply:
it's extended from that with our own implementation, howard can provide more info on this